lib/promscrape/client.go

```diff
@@ -95,7 +95,9 @@ func newClient(ctx context.Context, sw *ScrapeWork) (*client, error) {
 	setProxyHeaders := func(req *http.Request) error { return nil }
 	setFasthttpProxyHeaders := func(req *fasthttp.Request) error { return nil }
 	proxyURL := sw.ProxyURL
-	if !isTLS && proxyURL.IsHTTPOrHTTPS() {
+	// If it's not TLS, and the Proxy URL is HTTP(s), and the user didn't request to force HTTP CONNECT Passthrough. The
+	// latter check is to support konnectivity, which only uses HTTP CONNECT passthrough, even for non-TLS requests. #5109
```

Comment:
Could you please place a full link to an issue here?
It is usually easier to follow the link later on.

Reply:
done!